src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -170,35 +170,28 @@ private void OnEventSourceChanged(object? _1, TaggerEventArgs _2)
 
             private void EnqueueWork(bool initialTags)
             {
-                lock (_cancellationSeries)
+                using var stateRef = _tagSourceState.TryAddReference();
+
+                // No point proceeding if we've been disposed.
+                if (stateRef is null)
+                    return;
+
+                var state = stateRef.Target;
+
+                // cancel the last piece of computation work and enqueue the next.  This doesn't apply for the very
+                // first tag request we make.  We don't want that to be cancellable as we want that result to be 
+                // shown as soon as possible.
```

Comment:
Random nit: I do wish this comment was in GetCancellationToken; at least the order GitHub presented the diff in meant it was a bit confusing why we had an extra token at first. Once I saw this the it made sense.

Reply:
done.